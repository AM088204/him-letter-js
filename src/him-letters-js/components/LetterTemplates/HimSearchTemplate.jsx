import React from 'react';
import SearchField from 'terra-search-field';
import Text from 'terra-text';
import { injectIntl } from 'react-intl';
import classNames from 'classnames/bind';
import styles from '../../stylesheets/HimLetterTemplates.css';

const cx = classNames.bind(styles);
//Search bar to search templates.
const HimSearchTemplate = (props) => {

    const { intl } = props;

    return (
        <div className={cx(['margin-top'])}>
            <Text className={cx(['text-style'])}>{intl.formatMessage({ id: 'him-letters-engine.himSearchTemplate.label' })}</Text>
            <SearchField
                placeholder={intl.formatMessage({ id: 'him-letters-engine.himSearchTemplate.placeholder' })}
            />
        </div>
    );
}

export default injectIntl(HimSearchTemplate);