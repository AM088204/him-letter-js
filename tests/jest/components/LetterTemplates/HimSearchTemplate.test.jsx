import React from 'react';
import SearchField from 'terra-search-field';
import { shallowWithIntl, mountWithIntl } from 'terra-enzyme-intl';
import HimSearchTemplate from '../../../../src/him-letters-js/components/LetterTemplates/HimSearchTemplate';

describe('HimSearchTemplate', () => {
    it('Should Render HimSearchTemplate', () => {
        const himSearchTemplate = shallowWithIntl(<HimSearchTemplate />);
        expect(himSearchTemplate.first().shallow()).toMatchSnapshot();
    });

    it('should verify the attributes of the component', () => {
        const himSearchTemplate = mountWithIntl(
            <HimSearchTemplate />,
        );
        expect(himSearchTemplate.find(SearchField).props().placeholder).toBe('him-letters-engine.himSearchTemplate.placeholder');
    });

    it('Should verify the label', () => {
        const himSearchTemplate = mountWithIntl(
            <HimSearchTemplate />
        );
        expect(himSearchTemplate.contains('him-letters-engine.himSearchTemplate.label')).toBe(true);
    });
}
); 