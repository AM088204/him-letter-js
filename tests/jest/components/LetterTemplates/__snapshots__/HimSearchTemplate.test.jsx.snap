// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HimSearchTemplate Should Render HimSearchTemplate 1`] = `
<div
  className="margin-top"
>
  <Text
    className="text-style"
    isItalic={false}
    isVisuallyHidden={false}
    isWordWrapped={false}
  >
    him-letters-engine.himSearchTemplate.label
  </Text>
  <InjectIntl(SearchField)
    placeholder="him-letters-engine.himSearchTemplate.placeholder"
  />
</div>
`;
